### PR TITLE
POTEL 50 - Auto config for Spring Boot combined with OTel but without agent

### PR DESCRIPTION
**Pull request by adinauer**
## :pencil: Checklist

- [ ] I reviewed the submitted code.
- [ ] I added tests to verify the changes.
- [ ] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled.
- [ ] I updated the docs if needed.
- [ ] Review from the native team if needed.
- [ ] No breaking change or entry added to the changelog.
- [ ] No breaking change for hybrid SDKs or communicated to hybrid SDKs.

### CODE REVIEW

File: buildSrc/src/main/java/Config.kt [hunk 1]
```diff
@@ -98,6 +98,7 @@ object Config {
         val springBoot3StarterSecurity = "org.springframework.boot:spring-boot-starter-security:$springBoot3Version"
         val springBoot3StarterJdbc = "org.springframework.boot:spring-boot-starter-jdbc:$springBoot3Version"
         val springBoot3StarterActuator = "org.springframework.boot:spring-boot-starter-actuator:$springBoot3Version"
+        val springBoot3StarterOpenTelemetry = "io.opentelemetry.instrumentation:opentelemetry-spring-boot-starter:${OpenTelemetry.otelJavaagentVersion}"
```
Comment (lbloder) [inline buildSrc/src/main/java/Config.kt:101]: maybe we could rename that to `OpenTelemetry.otelInstrumentationVersion` as this no longer only relates to the agent

File: sentry-opentelemetry/sentry-opentelemetry-agentcustomization/src/main/java/io/sentry/opentelemetry/SentryAutoConfigurationCustomizerProvider.java [hunk 2]
```diff
@@ -29,31 +27,27 @@
 public final class SentryAutoConfigurationCustomizerProvider
     implements AutoConfigurationCustomizerProvider {
 
+  public static volatile boolean skipInit = false;
+
   @Override
   public void customize(AutoConfigurationCustomizer autoConfiguration) {
+    System.out.println("hello from agent");
```
Comment (lbloder) [inline sentry-opentelemetry/sentry-opentelemetry-agentcustomization/src/main/java/io/sentry/opentelemetry/SentryAutoConfigurationCustomizerProvider.java:34]: can be removed

File: sentry-opentelemetry/sentry-opentelemetry-extra/src/main/java/io/sentry/opentelemetry/OtelSpanFactory.java [hunk 3]
```diff
@@ -145,7 +157,14 @@ public final class OtelSpanFactory implements ISpanFactory {
   }
 
   private @NotNull Tracer getTracer() {
-    return GlobalOpenTelemetry.getTracer(
-        "sentry-instrumentation-scope-name", "sentry-instrumentation-scope-version");
+    return getTracerProvider().get("sentry-opentelemetry", BuildConfig.VERSION_NAME);
+  }
+
+  private @NotNull TracerProvider getTracerProvider() {
+    System.out.println("hello from " + toString());
```
Comment (lbloder) [inline sentry-opentelemetry/sentry-opentelemetry-extra/src/main/java/io/sentry/opentelemetry/OtelSpanFactory.java:164]: can be removed

File: sentry-samples/sentry-samples-spring-boot-jakarta-opentelemetry/build.gradle.kts [hunk 4]
```diff
@@ -52,7 +52,11 @@ dependencies {
     implementation(projects.sentryLogback)
     implementation(projects.sentryGraphql22)
     implementation(projects.sentryQuartz)
-    implementation(Config.Libs.OpenTelemetry.otelSdk)
+//    implementation(Config.Libs.OpenTelemetry.otelSdk)
```
Comment (lbloder) [inline sentry-samples/sentry-samples-spring-boot-jakarta-opentelemetry/build.gradle.kts:55]: can probably be removed

File: sentry-samples/sentry-samples-spring-boot-jakarta-opentelemetry/build.gradle.kts [hunk 5]
```diff
@@ -52,7 +52,11 @@ dependencies {
...
+    implementation("io.opentelemetry.instrumentation:opentelemetry-spring-boot-starter")
+    implementation(Config.Libs.springBoot3StarterOpenTelemetry)
```
Comment (lbloder) [inline sentry-samples/sentry-samples-spring-boot-jakarta-opentelemetry/build.gradle.kts:57]: are these the same dependencies?

File: sentry-samples/sentry-samples-spring-boot-jakarta-opentelemetry/src/main/java/io/sentry/samples/spring/boot/jakarta/CustomJob.java [hunk 6]
```diff
@@ -18,7 +17,7 @@ public class CustomJob {
   private static final Logger LOGGER = LoggerFactory.getLogger(CustomJob.class);
 
   @SentryCheckIn("monitor_slug_1")
-  @Scheduled(fixedRate = 3 * 60 * 1000L)
+  //  @Scheduled(fixedRate = 3 * 60 * 1000L)
```
Comment (lbloder) [inline sentry-samples/sentry-samples-spring-boot-jakarta-opentelemetry/src/main/java/io/sentry/samples/spring/boot/jakarta/CustomJob.java:20]: Could be re-enabled

File: sentry-samples/sentry-samples-spring-boot-jakarta-opentelemetry/src/main/java/io/sentry/samples/spring/boot/jakarta/PersonController.java [hunk 7]
```diff
@@ -19,19 +20,27 @@
...
-    this.tracer = tracer;
+    this.openTelemetry = openTelemetry;
   }
 
   @GetMapping("{id}")
+  @WithSpan("personSpanThroughOtelAnnotation")
   Person person(@PathVariable Long id) {
-    Span span = tracer.spanBuilder("spanCreatedThroughOtelApi").startSpan();
+    ISpan annotationSpan = Sentry.getSpan();
+    System.out.println(annotationSpan);
```
Comment (lbloder) [inline sentry-samples/sentry-samples-spring-boot-jakarta-opentelemetry/src/main/java/io/sentry/samples/spring/boot/jakarta/PersonController.java:35]: I believe these were just here for testing purposes, iirc

File: sentry-samples/sentry-samples-spring-boot-jakarta-opentelemetry/src/main/java/io/sentry/samples/spring/boot/jakarta/SentryDemoApplication.java [hunk 8]
```diff
@@ -42,37 +31,39 @@ RestClient restClient(RestClient.Builder builder) {
...
+  //    return trigger;
+  //  }
+  //
+  //  @Bean
+  //  public CronTriggerFactoryBean cronTrigger(JobDetail job) {
+  //    CronTriggerFactoryBean trigger = new CronTriggerFactoryBean();
+  //    trigger.setJobDetail(job);
+  //    trigger.setCronExpression("0 0/5 * ? * *"); // every five minutes
+  //    return trigger;
+  //  }
```
Comment (lbloder) [inline sentry-samples/sentry-samples-spring-boot-jakarta-opentelemetry/src/main/java/io/sentry/samples/spring/boot/jakarta/SentryDemoApplication.java:61]: Can be re-enabled

File: sentry-samples/sentry-samples-spring-boot-jakarta-opentelemetry/src/main/java/io/sentry/samples/spring/boot/jakarta/SentryDemoApplication.java [hunk 9]
```diff
@@ -42,37 +31,39 @@ RestClient restClient(RestClient.Builder builder) {
...
-  @Bean
-  public Tracer getTracer() {
-    return GlobalOpenTelemetry.getTracer("tracerForSpringBootDemo");
-  }
+  //  @Bean
+  //  public Tracer getTracer(OpenTelemetry openTelemetry) {
+  //    GlobalOpenTelemetry.set(openTelemetry);
+  //    return openTelemetry.getTracer("tracerForSpringBootDemo");
+  ////    return GlobalOpenTelemetry.getTracer("tracerForSpringBootDemo");
+  //  }
```
Comment (lbloder) [inline sentry-samples/sentry-samples-spring-boot-jakarta-opentelemetry/src/main/java/io/sentry/samples/spring/boot/jakarta/SentryDemoApplication.java:68]: Should this be re-enabled or removed?

File: sentry-samples/sentry-samples-spring-boot-jakarta/src/main/java/io/sentry/samples/spring/boot/jakarta/CustomJob.java [hunk 10]
```diff
@@ -18,7 +17,7 @@ public class CustomJob {
   private static final Logger LOGGER = LoggerFactory.getLogger(CustomJob.class);
 
   @SentryCheckIn("monitor_slug_1")
-  @Scheduled(fixedRate = 3 * 60 * 1000L)
+  //  @Scheduled(fixedRate = 3 * 60 * 1000L)
```
Comment (lbloder) [inline sentry-samples/sentry-samples-spring-boot-jakarta/src/main/java/io/sentry/samples/spring/boot/jakarta/CustomJob.java:20]: can be re-enabled

File: sentry-samples/sentry-samples-spring-boot-jakarta/src/main/java/io/sentry/samples/spring/boot/jakarta/PersonController.java [hunk 11]
```diff
@@ -13,20 +19,49 @@
...
   private static final Logger LOGGER = LoggerFactory.getLogger(PersonController.class);
 
-  public PersonController(PersonService personService) {
+  public PersonController(PersonService personService, Tracer tracer) {
     this.personService = personService;
+    this.tracer = tracer;
   }
 
   @GetMapping("{id}")
+  //  @WithSpan("personSpanThroughOtelAnnotation")
```
Comment (lbloder) [inline sentry-samples/sentry-samples-spring-boot-jakarta/src/main/java/io/sentry/samples/spring/boot/jakarta/PersonController.java:31]: can this be removed?

File: sentry-samples/sentry-samples-spring-boot-jakarta/src/main/java/io/sentry/samples/spring/boot/jakarta/SentryDemoApplication.java [hunk 12]
```diff
@@ -40,32 +33,37 @@ RestClient restClient(RestClient.Builder builder) {
...
+  //    return trigger;
+  //  }
+  //
+  //  @Bean
+  //  public CronTriggerFactoryBean cronTrigger(JobDetail job) {
+  //    CronTriggerFactoryBean trigger = new CronTriggerFactoryBean();
+  //    trigger.setJobDetail(job);
+  //    trigger.setCronExpression("0 0/5 * ? * *"); // every five minutes
+  //    return trigger;
+  //  }
```
Comment (lbloder) [inline sentry-samples/sentry-samples-spring-boot-jakarta/src/main/java/io/sentry/samples/spring/boot/jakarta/SentryDemoApplication.java:63]: Can be re-enabled

File: sentry-samples/sentry-samples-spring-boot-jakarta-opentelemetry/build.gradle.kts (hunk 4 above)
Comment (adinauer) [reply sentry-samples/sentry-samples-spring-boot-jakarta-opentelemetry/build.gradle.kts:55]: updated in a follow up PR

File: sentry-samples/sentry-samples-spring-boot-jakarta-opentelemetry/src/main/java/io/sentry/samples/spring/boot/jakarta/PersonController.java (hunk 7 above)
Comment (adinauer) [reply sentry-samples/sentry-samples-spring-boot-jakarta-opentelemetry/src/main/java/io/sentry/samples/spring/boot/jakarta/PersonController.java:35]: updated in a follow up PR

File: sentry-samples/sentry-samples-spring-boot-jakarta-opentelemetry/src/main/java/io/sentry/samples/spring/boot/jakarta/SentryDemoApplication.java (hunk 9 above)
Comment (adinauer) [reply sentry-samples/sentry-samples-spring-boot-jakarta-opentelemetry/src/main/java/io/sentry/samples/spring/boot/jakarta/SentryDemoApplication.java:68]: No, there's different code we want. updated in a follow up PR

File: sentry-samples/sentry-samples-spring-boot-jakarta/src/main/java/io/sentry/samples/spring/boot/jakarta/PersonController.java (hunk 11 above)
Comment (adinauer) [reply sentry-samples/sentry-samples-spring-boot-jakarta/src/main/java/io/sentry/samples/spring/boot/jakarta/PersonController.java:31]: updated in a follow up PR

File: sentry-samples/sentry-samples-spring-boot-jakarta/src/main/java/io/sentry/samples/spring/boot/jakarta/SentryDemoApplication.java (hunk 12 above)
Comment (adinauer) [reply sentry-samples/sentry-samples-spring-boot-jakarta/src/main/java/io/sentry/samples/spring/boot/jakarta/SentryDemoApplication.java:63]: updated in a follow up PR

File: sentry-opentelemetry/sentry-opentelemetry-agentcustomization/src/main/java/io/sentry/opentelemetry/SentryAutoConfigurationCustomizerProvider.java (hunk 2 above)
Comment (adinauer) [reply sentry-opentelemetry/sentry-opentelemetry-agentcustomization/src/main/java/io/sentry/opentelemetry/SentryAutoConfigurationCustomizerProvider.java:34]: updated in a follow up PR

File: sentry-opentelemetry/sentry-opentelemetry-extra/src/main/java/io/sentry/opentelemetry/OtelSpanFactory.java (hunk 3 above)
Comment (adinauer) [reply sentry-opentelemetry/sentry-opentelemetry-extra/src/main/java/io/sentry/opentelemetry/OtelSpanFactory.java:164]: updated in a follow up PR

File: buildSrc/src/main/java/Config.kt (hunk 1 above)
Comment (adinauer) [reply buildSrc/src/main/java/Config.kt:101]: updated in a follow up PR

**Review by lbloder — APPROVED** [review]

**Pull request merged**